app/scripts/controllers/metametrics.js

```diff
@@ -603,22 +613,30 @@ export default class MetaMetricsController {
   }
 
   /**
+   * Returns an array of all of the collectible/NFT addresses the user
+   * possesses across all networks and accounts.
    *
-   * @param {object} metamaskState
-   * @returns number of unique collectible addresses
+   * @param {Object} allCollectibles
+   * @returns {[]}
    */
-  _getNumberOfNFtCollection(metamaskState) {
-    const { allCollectibles } = metamaskState;
-    if (!allCollectibles) {
-      return 0;
-    }
-
-    const allAddresses = Object.values(allCollectibles)
+  _getAllNFTAddresses = memoize((allCollectibles = {}) => {
+    return Object.values(allCollectibles)
       .flatMap((chainCollectibles) => Object.values(chainCollectibles))
       .flat()
       .map((collectible) => collectible.address);
-    const unique = [...new Set(allAddresses)];
-    return unique.length;
+  });
+
+  /**
+   * Returns the number of unique collectible/NFT addresses the user
+   * possesses across all networks and accounts.
+   *
+   * @param {Object} allCollectibles
+   * @returns {number}
+   */
+  _getAllUniqueNFTAddressesLength(allCollectibles = {}) {
+    const allAddresses = this._getAllNFTAddresses(allCollectibles);
+    const uniqueAddresses = new Set(allAddresses);
+    return uniqueAddresses.size;
   }
```

Comment:
I don't think this is correct...A user may have multiple nfts from the same collection (address) @adonesky1 is that correct?

Reply:
Happy to jump on a call @digiwand and clarify these designations if that would help!

Reply:
Makes sense. I think this code is still correct. Maybe there is confusion because updates to both `number_of_nft_collections` and `number_of_nfts` are made in this PR. When I created this logic, I was going off of 

> All collectibles has this shape:
allCollectibles: { [key: string (account)]: { [key: string (chainId)]: Collectible[] } };

> This is an object whose keys are accounts and the values are objects where the keys are chainIds. So, for example, in order to get an aggregate number of NFTs imported you'll have to go through each chainId in each account in this object and sum the length of the Collectibles array.

> In order to get the number of collections you'll need to do the same sort of thing but you'll want the number of unique 'address' properties on the entries. Each collectible has an address property that indicates which collection it is a part of.

from the description in #13477. What you two mentioned sounds aligned with this description, and I think the code has been designed for this. 

This particular method called out here is used to compute the value for `number_of_nft_collections`, while `number_of_nfts` is using 
```
this._getAllNFTAddresses(
    metamaskState.allCollectibles,
).length,
```

After mentioning this, do you two still think we need to update the code? Maybe I'm missing something. Thank you! I also love hopping on calls with you two to pick your brains! 😁  

@brad-decker @adonesky1

Reply:
Sorry @digiwand -- i misread the code changes. So this would increment for each address regardless of whether it was unique. I conflated the changes with collections here.

Reply:
Ah ok. Yes this is correct. Though I think the method _getAllNFTAddresses is a bit of a confusing abstraction even if it is technically working to help get these two traits. I think maybe instead this could be called _getAllNFTs and it could be an array of tuples [address, tokenId], and then _getAllUniqueNFTAddressesLength would filter it down to an array of unique addresses in this array of tuples. Does that make sense?

Reply:
@brad-decker np! easy mix-up. I think it could have helped if I mentioned this in the description too. Updated the description 👍  

@adonesky1 I like this! Thanks. Update the code with your suggestion here: https://github.com/MetaMask/metamask-extension/pull/14495/commits/57b2bc62fe8891bcb753b1c5aeae35b90188dcc2